File: sunburst.php
<?php

if (in_array($argv[1], array('--help', '-help', '-h', '-?'))) {
?>

Vector Sunburst Generator - Digital Phantom

  Usage:
  <?php echo $argv[0]; ?> [file] [sides] [radius]

  [file] - file name where SVG file will be saved
  [sides] - the number of rays the sunburst will have
  [radius] - radius of the sunburst

(c) 2013 Digital Phantom - http://www.digitalphantom.net/
authored by: Yoel Nunez http://www.yoelnunez.com/
<?php
	exit;
}
else if($argc < 4) {
	exit("\nERROR: Pleasy type {$argv[0]} -h for help.\n\n");
}

$filename = $argv[1];
$sides = $argv[2];
$radius = $argv[3];

if(trim($filename) == "") {
	exit("\nERROR: filename cannot be please refer to the usage details for instructions.\n\n");
} else if(!is_numeric($sides)) {
	exit("\nERROR: The number of sides \"{$sides}\" has to be an integer value.\n\n");
}
else if(!is_numeric($radius)) {
	$radius = 50;
}

$points = 5 * $sides;

$angle = 360 / $points;

$coordinates = "";

for($i = 0; $i < $points; $i++) {
	$ix = sin(deg2rad($i * $angle)) * $radius;
	$iy = cos(deg2rad($i * $angle)) * - 1 * $radius;

	$x = $radius + $ix;
	$y = $radius + $iy;

	if($i % 5 == 0)	{
		$x = $y = $radius;
	}

	$coordinates .= "{$x},{$y} ";
}

$handle = fopen($filename, "w+");

if($handle) {
	fwrite($handle, '<svg xmlns="http://www.w3.org/2000/svg" version="1.1">');
	fwrite($handle, '<polygon points="'.trim($coordinates).'" style="fill:#000;"/>');
	fwrite($handle, '</svg>');
	fclose($handle);

	echo "\nYour vector file \"{$filename}\" was succesfully generated.\n\n";
}
else {
	echo "\nERROR: The file \"{$filename}\" could not be opened.\n\n";
}

File: sunburst.py
import sys
import math

args = sys.argv;

if len(args) >= 2 and args[1] == "-h":
    print("\n\nVector Sunburst Generator - Digital Phantom\n\n"
          "\tUsage:\n\t%s [file] [sides] [radius]\n\n"
          "\t[file] - file name where SVG file will be saved\n"
          "\t[sides] - the number of rays the sunburst will have\n"
          "\t[radius] - radius of the sunburst\n\n"
          "(c) 2013 Digital Phantom - http://www.digitalphantom.net/\n"
          "authored by: Yoel Nunez http://www.yoelnunez.com/" % args[0])
    exit()
elif len(args) < 4:
    print("\nERROR: Please type %s -h for help\n\n" % args[0])
    exit()

filename = args[1]
sides = int(args[2])
radius = int(args[3])

if str == "":
    print("\nERROR: filename cannot be empty, please refer to the usage details for instructons\n\n")
    exit()
elif sides == 0:
    print("\nERROR: the number of sides \"%d\" has to be an integer > 0\n\n" % sides)
    exit()
elif radius == 0:
    radius = 50

points = 5 * sides

angle = 360 / points

coordinates = ""

for i in range(points):
    ix = math.sin(math.radians(i * angle)) * radius
    iy = math.cos(math.radians(i * angle)) * - 1 * radius

    x = radius + ix
    y = radius + iy

    if i % 5 == 0:
        x = y = radius

    coordinates += str(x) + "," + str(y) + " "

try:
    handle = open(filename, "w+")

    handle.write('<svg xmlns="http://www.w3.org/2000/svg" version="1.1">')
    handle.write('<polygon points="' + coordinates + '" style="fill:#000;"/>')
    handle.write('</svg>')

    print("\nYour vector file \"%s\" was successfully generated.\n\n" % filename)

except IOError:
    print("\nERROR: The file \"%s\" could not be opened.\n\n" % filename);